TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/stateMachine/StateMachineOpmode.java

```diff
@@ -0,0 +1,252 @@
+        if-statement as the initial "Team Prop" scoring opportunities. */
+        if (!StateMachineOpmode.teamPropLineScored || !StateMachineOpmode.teamPropBackdropScored) {
+            if (!StateMachineOpmode.teamPropLineScored) {
+                driveToTeamPropLine();
+            } else {
+                driveToBackdrop();
+            }
+        } else if (StateMachineOpmode.elapsedTime.time() > 25) {
+            driveToAutonPark();
+        } else if (StateMachineOpmode.pixelsInControl < 2) {
+            /* Sets state to scoring. */
+            driveToIntakePixels();
+        } else if (StateMachineOpmode.pixelsInControl == 2) {
+            /* Sets state to scoring. */
+            driveToBackdrop();
+        } else {
+            /* Not throwing RuntimeException to avoid crashing robot. This code should
+            theoretically never be run, but if it is, it does not necessarily indicate a serious
+            error, just something that we should be aware of to avoid silent logic errors.  */
+            telemetry.addLine("Current status unaccounted for in Drive State.");
+            telemetry.addData("time", StateMachineOpmode.elapsedTime);
+            telemetry.addData("teamPropLineScored", StateMachineOpmode.teamPropLineScored);
+            telemetry.addData("TeamPropBackdropScored", StateMachineOpmode.teamPropBackdropScored);
+            telemetry.addData("controlledPixels", StateMachineOpmode.teamPropLineScored);
+        }
+    }
+
+    public void driveToBackdrop() {
+        telemetry.addLine("Driving to Backdrop.");
+
+        // TODO: Make this method actually make the robot drive to the backdrop...
+        StateMachineOpmode.inScoringPosition = true;
+    }
+
+    public void driveToTeamPropLine() {
+        telemetry.addLine("Driving to Team Prop Line.");
+
+        // TODO: Make the robot drive to the Team Prop line
+        /* "Scoring" State generalized to handle scoring on the Team Prop Line and on the backdrop. */
+        StateMachineOpmode.inScoringPosition = true;
+    }
+
+    public void driveToIntakePixels() {
+        telemetry.addLine("Driving to pixels.");
+
+        // TODO: Make the robot figure out which pixels to go to, and then go to them
+        StateMachineOpmode.inIntakePosition = true;
+    }
+
+    public void driveToAutonPark() {
+        telemetry.addLine("Parking.");
+
+        // TODO: Make robot Auton Park
+        StateMachineOpmode.robotParked = true;
+    }
+
+}
+
+class ScoringState implements State {
+    ArrayList<Edge> edges = new ArrayList();
+
+    public ArrayList<Edge> getEdges() {
+        edges.add(new Edge(IdleState.class, () -> StateMachineOpmode.robotParked));
+        edges.add(new Edge(DrivingState.class, () -> StateMachineOpmode.pixelsInControl == 0));
+
+        return edges;
+    }
+
+    public void loop() {
+        telemetry.addData("State", "Driving State");
+
+        if (!StateMachineOpmode.teamPropLineScored) {
```

Comment:
If you have to do this sort of logic, they should be separated into distinct states, probably with shared functionality.